src/app/CommandHandler.cpp

```diff
@@ -247,6 +248,21 @@ CHIP_ERROR CommandHandler::ProcessCommandDataIB(CommandDataIB::Parser & aCommand
     VerifyOrExit(mpCallback->CommandExists(ConcreteCommandPath(endpointId, clusterId, commandId)),
                  err = CHIP_ERROR_INVALID_PROFILE_ID);
 
+    {
+        Access::SubjectDescriptor subjectDescriptor; // TODO: get actual subject descriptor
+        Access::RequestPath requestPath{ .cluster = clusterId, .endpoint = endpointId };
+        Access::Privilege requestPrivilege = Access::Privilege::kOperate; // TODO: get actual request privilege
+        err                                = Access::GetAccessControl().Check(subjectDescriptor, requestPath, requestPrivilege);
+        if (err != CHIP_NO_ERROR)
+        {
+            const ConcreteCommandPath concretePath(endpointId, clusterId, commandId);
+            auto status = (err == CHIP_ERROR_ACCESS_DENIED) ? Protocols::InteractionModel::Status::UnsupportedAccess
+                                                            : Protocols::InteractionModel::Status::Failure;
+            AddStatus(concretePath, status);
```

Comment:
Does the spec define the order of operations between parsing the command data and doing this check?  If not, please file a spec issue (to either define or explicitly say it's not defined).

Reply:
Spec issue #4744